crates/router/src/connector/prophetpay/transformers.rs

```diff
@@ -22,122 +28,339 @@ impl<T>
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.router_data.request.is_auto_capture()?,
-                };
+            api::PaymentMethodData::CardRedirect(
+                api_models::payments::CardRedirectData::CardRedirect {},
+            ) => {
+                let auth_data =
+                    ProphetpayAuthType::try_from(&item.router_data.connector_auth_type)?;
                 Ok(Self {
-                    amount: item.amount.to_owned(),
-                    card,
+                    ref_info: item.router_data.connector_request_reference_id.to_owned(),
+                    profile: auth_data.profile_id,
+                    entry_method: get_entry_method(ProphetpayEntryMethod::ManualEntry),
+                    token_type: get_token_type(ProphetpayTokenType::SaleTab),
+                    card_entry_context: get_card_context(
+                        ProphetpayCardContext::WebConsumerInitiated,
+                    ),
                 })
             }
             _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
         }
     }
 }
 
-pub struct ProphetpayAuthType {
-    pub(super) api_key: Secret<String>,
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ProphetpayTokenResponse {
+    hosted_tokenize_id: String,
 }
 
-impl TryFrom<&types::ConnectorAuthType> for ProphetpayAuthType {
+impl<F>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            ProphetpayTokenResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::PaymentsAuthorizeData, types::PaymentsResponseData>
+{
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
-        match auth_type {
-            types::ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
-                api_key: api_key.to_owned(),
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            ProphetpayTokenResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let url_data = format!(
+            "https://ccm-thirdparty.cps.golf/hp/tokenize/{}",
+            item.response.hosted_tokenize_id
+        );
+
+        let redirect_url = Url::parse(url_data.as_str())
+            .into_report()
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+
+        let redirection_data = get_redirect_url_form(
+            redirect_url,
+            item.data.request.complete_authorize_url.clone(),
+        )
+        .ok();
+
+        Ok(Self {
+            status: enums::AttemptStatus::AuthenticationPending,
```

Comment:
This is not authentication pending. Set it to Started

Reply:
are not we doing verification in the redirection? and setting it to start will make the status processing because of which we wont be getting redirection link